src/Features/CSharp/Portable/DocumentationComments/DocumentationCommentSnippetService.cs

```diff
@@ -111,22 +124,108 @@ protected override List<string> GetDocumentationCommentStubLines(MemberDeclarati
                 }
             }
 
-            if (member.IsKind(SyntaxKind.MethodDeclaration) ||
-                member.IsKind(SyntaxKind.IndexerDeclaration) ||
-                member.IsKind(SyntaxKind.DelegateDeclaration) ||
-                member.IsKind(SyntaxKind.OperatorDeclaration))
+            if (member.IsKind(
+                    SyntaxKind.MethodDeclaration,
+                    SyntaxKind.IndexerDeclaration,
+                    SyntaxKind.DelegateDeclaration,
+                    SyntaxKind.OperatorDeclaration,
+                    SyntaxKind.ConstructorDeclaration,
+                    SyntaxKind.DestructorDeclaration))
             {
                 var returnType = member.GetMemberType();
                 if (returnType != null &&
                     !(returnType.IsKind(SyntaxKind.PredefinedType, out PredefinedTypeSyntax? predefinedType) && predefinedType.Keyword.IsKindOrHasMatchingText(SyntaxKind.VoidKeyword)))
                 {
                     list.Add("/// <returns></returns>");
                 }
+
+                foreach (var exceptionType in GetExceptions(member, model, options, cancellationToken))
+                {
+                    list.Add(@$"/// <exception cref=""{exceptionType}""></exception>");
+                }
             }
 
             return list;
         }
 
+        private static IEnumerable<string> GetExceptions(SyntaxNode member, SemanticModel model, OptionSet options, CancellationToken cancellationToken)
+        {
+            var throwExpressionsAndStatements = member.DescendantNodes().Where(n => n.IsKind(SyntaxKind.ThrowExpression, SyntaxKind.ThrowStatement));
+            var namespacesInScope = model.GetUsingNamespacesInScope(member);
+            // We generate <exception cref="System.NullReferenceException"> if there is no 'using System;', and generate
+            // <exception cref="NullReferenceException"> if there is a 'using System;'
```

Comment:
this feels unnecessary.  we can just look at how the `throw` looked.  if teh user says `throw NullReferenceException` then we generate the same.  if they say `throw System.NullReferenceException` likewise.